method/method.go

```diff
@@ -229,21 +237,37 @@ func newLocation(value string) (objectLocation, error) {
 	if err != nil {
 		return objectLocation{}, nil
 	}
-	tokens := strings.Split(uri.Path, "/")
+	if uri.Host == s3Hostname {
```

Comment:
Just to be clear, the problem being solved here is supporting bucket uri's with the form `private-repo.s3.amazonaws.com` vs `s3.amazonaws.com/private-repo`?

Reply:
Yes, now we support 3 types of s3 repo URL, s3://<bucket>, s3://<bucket>.s3.amazonaws.com besides current one.